contracts/Storage.sol

```diff
@@ -82,8 +85,17 @@ contract Storage is Collateral, Marketplace {
     slotMustAcceptProofs(slotId)
   {
     _markProofAsMissing(slotId, period);
+    address host = _host(slotId);
     if (_missed(slotId) % slashMisses == 0) {
-      _slash(_host(slotId), slashPercentage);
+        _slash(host, slashPercentage);
+
+      if (balanceOf(host) < minCollateralThreshold) {
+        // If host has been slashed enough such that the next slashing would
+        // cause the collateral to drop below the minimum threshold, the slot
+        // needs to be freed so that there is enough remaining collateral to be
+        // distributed for repairs and rewards (with any leftover to be burnt).
```

Comment:
```suggestion
        // When the collateral drops below the minimum threshold, the slot
        // needs to be freed so that there is enough remaining collateral to be
        // distributed for repairs and rewards (with any leftover to be burnt).
```